hvac/v1/__init__.py

```diff
@@ -77,16 +77,13 @@ def delete(self, path):
 
     def unwrap(self, token):
         """
-        GET /cubbyhole/response
+        POST /sys/wrapping/unwrap
         X-Vault-Token: <token>
         """
-        path = "cubbyhole/response"
-        _token = self.token
-        try:
-            self.token = token
-            return json.loads(self.read(path)['data']['response'])
-        finally:
-            self.token = _token
+        payload = {
+            'token': token
```

Comment:
Would you mind making this token parameter optional? As mentioned in the [Vault documentation](https://www.vaultproject.io/api/system/wrapping-unwrap.html) regarding this parameter:
> Specifies the wrapping token ID. This is required if the client token is not the wrapping token. Do not use the wrapping token in both locations.

So as-is (without the token being optional), this method can be used to unwrap a token provided the *Client's* (i.e., the `hvac.Client` object) token attribute is valid. The other use case an optional token parameter would support is when you don't have any authentication token to begin with and want to do something like:

```
client = Client(token='some-wrapped-token')
unwrap_response = client.unwrap()
new_client_token = unwrap_response['auth']['client_token']
client.token = client_token

```

Though now that I've written up that example, I imagine that use case could be covered by calling the method with `client.unwrap(token='')` and/or `client.unwrap(token=None)`.

What do you think?

Reply:
I think it would probably lead to less frustration if hvac inferred the location of the wrapped material should be instead of the user specifying it. The Vault API behaves similarly so no reason to add opacity. :)

I've made the changes, please would you take a look.